### PR TITLE
feat(stdlib): implement onEmpty parameter for filter

### LINKED ISSUE #2278: Update filter to add onEmpty parameter
#2220

### PR DESCRIPTION
**Pull request by jsternberg**
The `filter()` function now accepts an `onEmpty` parameter that will
determine the behavior of filter when the filter causes all rows to be
filtered.

Fixes #2278.

### Done checklist
- [x] docs/SPEC.md updated
- [x] Test cases written

### CODE REVIEW

File: stdlib/universe/filter_keep_empty_test.flux [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+#default,_result,,,,,,,,
+,result,table,_start,_stop,_measurement,_field,host,name,_value
+,,0,2018-05-22T19:53:26Z,2030-01-01T00:00:00Z,diskio,io_time,host.local,disk0,6
+,,1,2018-05-22T19:53:26Z,2030-01-01T00:00:00Z,diskio,io_time,host.local,disk2,0
+"
+
+t_filter_keep_empty = (table=<-) =>
+  table
+  |> range(start: 2018-05-22T19:53:26Z)
+  |> filter(fn: (r) => r["_value"] > 1000, onEmpty: "keep")
```
Comment (nathanielc) [inline stdlib/universe/filter_keep_empty_test.flux]: Can you add a test case that shows how you would use  a combination of filters to get the empty tables from a subset of the data? Meaning say I have data for both `mem` and `cpu` measurements. I want empty table but on the from the `mem` measurement.

Comment (jsternberg) [reply stdlib/universe/filter_keep_empty_test.flux]: Yes I think I can do that.

Comment (jsternberg) [reply stdlib/universe/filter_keep_empty_test.flux]: Done.

File: stdlib/universe/filter_mixed_empty_test.flux [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+,result,table,_start,_stop,_measurement,_field,host,name,_value
+,,0,2018-05-22T19:53:26Z,2030-01-01T00:00:00Z,diskio,io_time,host.local,disk0,6
+,,1,2018-05-22T19:53:26Z,2030-01-01T00:00:00Z,diskio,io_time,host.local,disk2,0
+"
+
+t_filter_mixed_empty = (table=<-) =>
+  table
+  |> range(start: 2018-05-22T19:53:26Z)
+  |> filter(fn: (r) => r._measurement == "diskio")
+  |> filter(fn: (r) => r["_value"] > 1000, onEmpty: "keep")
```
Comment (nathanielc) [inline stdlib/universe/filter_mixed_empty_test.flux]: Can you update this test case such that the output has an empty table? That way its a positive test that `keep` does what we expect.

Comment (jsternberg) [reply stdlib/universe/filter_mixed_empty_test.flux]: I actually can't. Diff doesn't detect the difference between empty tables and non-existant tables correctly. It considers them the same. This is likely a bug in diff, but I didn't want to start touching that. That's the reason why I use `count()` because it is a way to get around diff being unable to do this.

Comment (nathanielc) [reply stdlib/universe/filter_mixed_empty_test.flux]: OK, good to know, Can you create an issue fo the `diff` bug?

Comment (jsternberg) [reply stdlib/universe/filter_mixed_empty_test.flux]: https://github.com/influxdata/flux/issues/2291
